### PR TITLE
UI fixes

### PR DESCRIPTION
**Pull request by tritonas00**
- hide main menu while connecting
- fixed chat closing using esc
- made survey map character icon green again
- added restart notice in lights setting

### CODE REVIEW

File: source/main/gui/panels/GUI_GameSettings.cpp [hunk 1]
```diff
@@ -251,7 +251,7 @@ void GameSettings::Draw()
     {
         ImGui::TextDisabled(_LC("GameSettings", "Video settings"));
 
-        DrawGCombo(App::gfx_flares_mode, _LC("GameSettings", "Lights"),
+        DrawGCombo(App::gfx_flares_mode, _LC("GameSettings", "Lights (requires restart)"),
```
Comment (only-a-ptr) [inline source/main/gui/panels/GUI_GameSettings.cpp:254]: Not true. I tried returning to menu, setting lights to NONE and turning on headlights - the HUD indicator lit up, but the flares didn't appear.

File: source/main/gui/panels/GUI_GameChatBox.cpp [hunk 2]
```diff
@@ -118,7 +118,7 @@ void GameChatBox::Draw()
     ImGui::PushStyleColor(ImGuiCol_WindowBg, ImVec4(0,0.0,0,0)); // Fully transparent background!
     ImGui::Begin("ChatBottomBar", nullptr, chat_flags);
 
-    if (ImGui::IsItemActive() && ImGui::IsKeyPressed(ImGui::GetKeyIndex(ImGuiKey_Escape)))
+    if (ImGui::IsKeyPressed(ImGui::GetKeyIndex(ImGuiKey_Escape)))
```
Comment (only-a-ptr) [inline source/main/gui/panels/GUI_GameChatBox.cpp:121]: Looks good.

File: source/main/main.cpp [hunk 3]
```diff
@@ -367,6 +367,7 @@ int main(int argc, char *argv[])
 
                 case MSG_NET_CONNECT_PROGRESS:
                     App::GetGuiManager()->GetLoadingWindow()->SetProgressNetConnect(m.description);
+                    App::GetGuiManager()->SetVisible_GameMainMenu(false);
```
Comment (only-a-ptr) [inline source/main/main.cpp:370]: This is a hack, there's multiple _PROGRESS messages during connecting and no reason for main menu to appear. I'll investigate it why it appears at all.

Update: right, here's the actual problem: https://github.com/RigsOfRods/rigs-of-rods/blob/d129ba5a7df15f2d97fcc1a497ec36c41ce34771/source/main/gui/panels/GUI_MultiplayerSelector.cpp#L362-L365

File: source/main/gui/panels/GUI_SurveyMap.cpp [hunk 4]
```diff
@@ -221,7 +221,7 @@ void SurveyMap::Draw()
             if (!simbuf.simbuf_actor_coupling)
             {
                 std::string caption = (App::mp_state->GetEnum<MpState>() == MpState::CONNECTED) ? simbuf.simbuf_net_username : "";
-                this->DrawMapIcon(tl_screen_pos, view_size, view_origin, "icon_person.dds", caption, 
+                this->DrawMapIcon(tl_screen_pos, view_size, view_origin, "icon_person_activated.dds", caption,
```
Comment (only-a-ptr) [inline source/main/gui/panels/GUI_SurveyMap.cpp:224]: Looks good.

File: source/main/gui/panels/GUI_GameSettings.cpp (hunk 1 above)
Comment (tritonas00) [reply source/main/gui/panels/GUI_GameSettings.cpp:254]: If you don't restart RoR, it will crash (try the ant12 plane)

**Comment by only-a-ptr** [comment]
Superseded by #2608

**Pull request closed without merging**